exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/SimpleCastFunctions.java

```diff
@@ -48,14 +46,8 @@ public void setup() {
     public void eval() {
       byte[] buf = new byte[in.end - in.start];
       in.buffer.getBytes(in.start, buf, 0, in.end - in.start);
-      String input = new String(buf, com.google.common.base.Charsets.UTF_8).toLowerCase();
-      if ("true".equals(input)) {
-        out.value = 1;
-      } else if ("false".equals(input)) {
-        out.value = 0;
-      } else {
-        throw new IllegalArgumentException("Invalid value for boolean: " + input);
-      }
+      String input = new String(buf, com.google.common.base.Charsets.UTF_8);
+      out.value = org.apache.drill.exec.expr.BooleanType.get(input).getNumericValue();
```

Comment:
This \#get call is invoked per record.

How about storing a mapping (static `Map<String, Integer>`) in BooleanType, rather than iterate through a list?